libs/hdf-converters/src/converters-from-hdf/html/reverse-html-mapper.ts

```diff
@@ -201,7 +208,11 @@ export class FromHDFToHTMLMapper {
     const allResultLevels: ContextualizedControl[] = [];
     file.data.contains.map((profile) => {
       profile.contains.map((result) => {
-        allResultLevels.push(result);
+        file.filteredControls.forEach((element) => {
```

Comment:
Sure I can do that, however that is a personal preferences (map to flatMap), aside for the for...of which is an ES6, vs the forEach which is an ES5.

Reply:
map to flatMap is more than just a syntactic difference.

in any case, it's just a consistency in coding paradigm fix - we're doing functional style "loops" with the maps, but then all the way inside we do a for loop around a procedural line where we are populating an array defined outside of all of these loops.

imo we should either go full procedural and use actual loops for everything or go full functional instead of this mishmash.  my simplification recommendation was to do the functional solution.